### PR TITLE
Other set of fixes and improvements from old issues and stale PRs

### PR DESCRIPTION
**Pull request by filippotoso**

### CODE REVIEW

File: src/Checkers/User/LaratrustUserQueryChecker.php [hunk 1]
```diff
@@ -23,15 +23,7 @@ public function getCurrentUserRoles($team = null)
             return $this->user->roles->pluck('name')->toArray();
         }
 
-        if ($team === null) {
-            return $this->user
```
Comment (santigarcor) [inline src/Checkers/User/LaratrustUserQueryChecker.php:27]: Why did you remove this whole block?

**Comment by santigarcor** [comment]
Looks good to me, but I'm not sure sure if the domain thing works. From reading the code i think it does but haven't tested it yet.

Also, we should consider type hinting everything we can. Not for this PR

**Comment by filippotoso** [comment]
I tested locally the subdomain code and it seems to work perfectly.

I had to make some changes from the original proposal to allow backword compatibility for those that doesn't have the domain options set in the config file.

**Comment by santigarcor** [comment]
Approving this.

But we should start thinking on creating a new 8.x version so we can introduce breaking changes

**Review by santigarcor — APPROVED** [review]

**Comment by santigarcor** [comment]
BTW thank you for the help!
